app/Transformers/ScoreTransformer.php

```diff
@@ -102,6 +104,7 @@ public function transformSolo(MultiplayerScoreLink|ScoreModel|SoloScore $score)
         if ($score instanceof SoloScore) {
             $extraAttributes['ranked'] = $score->ranked;
             $extraAttributes['preserve'] = $score->preserve;
+            $extraAttributes['processed'] = $score->legacy_score_id !== null || $score->processHistory !== null;
```

Comment:
Uhhh... So earlier I said:

> > Is every score guaranteed to have a corresponding `score_process_history` row? Does `score_process_history` get truncated?
> 
> As far as I'm aware, yes.

So as it turns out, I lied here (or if you wanna be charitable, "forgot"). It is not always the case that a `score_process_history` row will exist for a score that is processed.

The exception is scores that have been initially manually batch-imported from `osu_scores_high_*` tables. All legacy scores generally are fully processed, because first `osu-web-10` processes them synchronously, and then the score watchers import them without doing any further processing relevant to the score - so at the point the `scores` row exists for a legacy score, it is already fully processed.

I can't exactly double-check production to confirm this but I realised this when testing on my local database wherein I had a local score import from seeded `osu_scores_high_*` rows into `scores`.

What this says about this particular PR, and also the complexity in and around the new score table, I'm not entirely sure, but I do acknowledge that it's pretty bad...